edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c

```diff
@@ -1595,22 +1596,36 @@ int generate_pki_cert_and_key_with_props
 int generate_pki_cert_and_key
 (
     CERT_PROPS_HANDLE cert_props_handle,
-    int serial_number,
     int ca_path_len,
     const char* key_file_name,
     const char* cert_file_name,
     const char* issuer_key_file,
     const char* issuer_certificate_file
 )
 {
-    return generate_pki_cert_and_key_helper(cert_props_handle,
+    int result;
+
+    long serial_number;
+    if (generate_rand_buffer((unsigned char*)&serial_number, sizeof(serial_number)) != 0)
+    {
+        LOG_ERROR("could not generate serial number");
+        result = __FAILURE__;
+    }
+    else
+    {
+        serial_number &= ((unsigned long)-1) >> 1;

```

Comment:
Making this a call to `labs()` is probably more readable.

Reply:
No, because it doesn't work for `LONG_MIN`.

Reply:
Oh right.

Reply:
Added a comment.